packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -507,13 +507,35 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return this.jsxParseElementAt(startPos, startLoc);
     }
 
+    // Validates and parses the current state as JSXText
+
+    parseJSXText(): N.Expression {
+      const startPos = this.state.start;
+      for (let idx = 0; idx < this.state.value.length; idx += 1) {
+        const character = this.state.value[idx];
+        if (character === "}") {
+          this.raise(
+            startPos + idx,
+            `Unexpected token \`}\`. Did you mean \`{'}'}\`?`,
+          );
+        } else if (character === ">") {
+          this.raise(
+            startPos + idx,
+            `Unexpected token \`>\`. Did you mean \`{'>'}\`?`,
+          );
+        }
+      }
+
+      return this.parseLiteral(this.state.value, "JSXText");
+    }
+
     // ==================================
     // Overrides
     // ==================================
 
     parseExprAtom(refExpressionErrors: ?ExpressionErrors): N.Expression {
       if (this.match(tt.jsxText)) {
-        return this.parseLiteral(this.state.value, "JSXText");
+        return this.parseJSXText();
```

Comment:
It seems I pointed you to the wrong location, sorry about that!

At this point, we've already tokenized the source into a `tt.jsxText` token (that's the `this.match()` check above). But we can do this earlier, when we're still creating the `jsxText` token. It's located at https://github.com/babel/babel/blob/1815c349877acca1d3275ea72512ee7824db49e3/packages/babel-parser/src/plugins/jsx/index.js#L81-L120.

Reply:
ezpz I'll fix that up tomorrow morning.
thanks for the pointers!

Reply:
ezpz I'll fix that up tomorrow morning.
thanks for the pointers!

Reply:
ezpz I'll fix that up tomorrow morning.
thanks for the pointers!